tidy3d/components/lumped_element.py

```diff
@@ -107,24 +140,148 @@ def _voltage_axis_in_plane(cls, val, values):
+        """Creates a suggested :class:`MeshOverrideStructure` list that could be added to the
+        :class:`Simulation`"""
+        # Create a mesh override when refinement is needed.
+        # An important use case is when a LumpedResistor is used with a LumpedPort
+        # The port is a flat surface, but when computing the port current,
+        # we'll eventually integrate the magnetic field just above and below
+        # this surface, so the mesh override needs to ensure that the mesh
+        # is fine enough not only in plane, but also in the normal direction.
+        # So in the normal direction, we'll make sure there are at least
+        # 2 cell layers above and below whose size is the same as the in-plane
+        # cell size in the override region. Also, to ensure that the port itself
+        # is aligned with a grid boundary in the normal direction, two separate
+        # override regions are defined, one above and one below the analytical
+        # port region.
+        mesh_overrides = []
+        size = 3 * [self.outer_diameter]
+        size[self.normal_axis] = 0
+        if self.num_grid_cells:
+            override_dl = [self.outer_diameter / self.num_grid_cells] * 3
+            override_dl[self.normal_axis] /= 2
+            override_size = size
+            override_size[override_size.index(0)] = 2 * override_dl[self.normal_axis]
+            override_center_above = list(self.center)
+            override_center_above[self.normal_axis] += override_dl[self.normal_axis]
+            override_center_below = list(self.center)
+            override_center_below[self.normal_axis] -= override_dl[self.normal_axis]
+            mesh_overrides.append(
+                MeshOverrideStructure(
+                    geometry=Box(center=override_center_below, size=override_size),
+                    dl=override_dl,
+                )
+            )
+            mesh_overrides.append(
+                MeshOverrideStructure(
+                    geometry=Box(center=override_center_above, size=override_size),
+                    dl=override_dl,
+                )
+            )
+        return mesh_overrides
+
+    @pydantic.validator("center", always=True)
+    def _center_not_inf(cls, val):
+        """Make sure center is not infinitiy."""
+        if any(np.isinf(v) for v in val):
+            raise ValidationError("center can not contain td.inf terms.")
+        return val
+
+    @pydantic.validator("inner_diameter", always=True)
+    @skip_if_fields_missing(["outer_diameter"])
+    def _ensure_inner_diameter_is_smaller(cls, val, values):
+        """Ensures that the inner diameter is smaller than the outer diameter, so that the final shape is an annulus."""
+        outer_diameter = values.get("outer_diameter")
+        if val >= outer_diameter:
+            raise ValidationError(
+                f"The inner diameter {val} of a coaxial lumped element must be less than the outer diameter {outer_diameter}."
```

Comment:
maybe let's use the variable names `inner_diameter` and `outer_diameter` (in single quotes)